manim/scene/scene.py

```diff
@@ -652,91 +653,105 @@ def get_moving_mobjects(self, *animations):
-            step = 1 / self.camera.frame_rate
-            times = np.arange(0, run_time, step)
-        time_progression = ProgressDisplay(
-            times, total=n_iterations,
-            leave=self.leave_progress_bars,
-            ascii=False if platform.system() != 'Windows' else True
+
+        return GlobalProgress(
+            "[progress.description]{task.description}",
+            BarColumn(),
+            "[progress.percentage]{task.completed}/{task.total}",
+            TimeRemainingColumn(),
         )
-        return time_progression
 
-    def get_run_time(self, animations):
+    def get_animation_description(self, animations):
         """
-        Gets the total run time for a list of animations.
+        Gets the progress bar description for a list of animations.
 
         Parameters
         ----------
         animations: list of Animation
-            A list of the animations whose total
-            run_time is to be calculated.
+            A list of the animations whose description will be returned.
 
         Returns
-        ------
-        float
-            The total run_time of all of the animations in the list.
+        -------
+        str
+            The progress bar description of the animations in the list.
         """
 
-        return np.max([animation.run_time for animation in animations])
+        description = f"Animation {self.num_plays}: {animations[0]}"
+        if len(animations) > 1:
+            description += ", etc."
+        return self.get_short_description(description)
+
+    def get_wait_description(self, stop_condition):
+        """
+        Gets the progress bar description for a wait.
 
-    def get_animation_time_progression(self, animations):
+        Parameters
+        ----------
+        stop_condition: Union[function, NoneType]
+            The condition under which the wait will stop or None if the wait
+            is based on time.
+
+        Returns
+        -------
+        str
+            The progress bar description of the wait.
         """
-        You will hardly use this when making your own animations.
-        This method is for Manim's internal use.
 
-        Uses get_time_progression to obtaina
-        CommandLine ProgressBar whose fill_time is
-        dependent on the qualities of the passed Animation,
+        if stop_condition is not None:
+            description = "Waiting for {}".format(stop_condition.__name__)
+        else:
+            description = "Waiting {}".format(self.num_plays)
+        return self.get_short_description(description)
+
+    def get_short_description(self, description):
+        """
+        Shortens a description to a maximum length of
+        self.max_description_width if necessary and pads it with ... if it was
+        shortened.
 
         Parameters
         ----------
-        animations : list of Animation
-            The list of animations to get
-            the time progression for.
+        description: str
```

Comment:
`` :class:`str` ``  (so sphinx renders it correctly)
```suggestion
        description: :class:`str`
```